src/main/java/igym/services/GymService.java

```diff
@@ -73,20 +75,25 @@ public List<Gym> findAllGyms() {
         return gyms;
     }
 
+    @Transactional
     public void deleteGym(UUID id) {
-        logger.info("Attempting to inactivate gym with id {}", id);
+        logger.info("Attempting to inactivate gym with id {}", id);        
         Gym gym = findById(id);
 
-        if (gym.getStatus() == Status.inactive) {
-            logger.warn("Gym with id {} is inactive", id);
-            throw new GymNotFoundException("Gym with id " + id + " not found");
-        }
-
+        inactivateWorkouts(gym);
         gym.setStatus(Status.inactive);
         gymRepository.save(gym);
         logger.info("Gym with id {} inactivated", id);
     }
-    
+
+    private void inactivateWorkouts(Gym gym) {
+        List<Workout> workouts = workoutRepository.findByGym(gym);
+        if(workouts != null) {
+            workouts.forEach(w -> workoutService.deleteWorkout(w.getId()));
+        }
+        workoutRepository.saveAll(workouts);
+    }
+
```

Comment:
there is an issue here, if the workout list is empty I don't think this gonna work, the shape of this method is a bit weird to me